index.js

```diff
@@ -8,37 +8,40 @@ module.exports = function fetchCookieDecorator (fetch, jar) {
   var getCookieString = denodeify(jar.getCookieString.bind(jar))
   var setCookie = denodeify(jar.setCookie.bind(jar))
 
-  return function fetchCookie (url, opts) {
+  async function fetchCookie (url, opts) {
     opts = opts || {}
 
-    return getCookieString(url)
-      .then(function (cookie) {
-        return fetch(url, Object.assign(opts, {
-          headers: Object.assign(opts.headers || {}, (cookie ? { cookie: cookie } : {}))
-        }))
-      })
-      .then(function (res) {
-        var cookies
-
-        if (res.headers.getAll) {
-          // node-fetch v1
-          cookies = res.headers.getAll('set-cookie')
-        } else {
-          // node-fetch v2
-          var cookie = res.headers.get('set-cookie')
-          // FIXME: This is not working. See issue #22 (https://github.com/valeriangalliat/fetch-cookie/issues/22)
-          cookies = cookie.split(',') || []
-        }
-
-        if (!cookies.length) {
-          return res
-        }
-
-        return Promise.all(cookies.map(function (cookie) {
-          return setCookie(cookie, res.url)
-        })).then(function () {
-          return res
-        })
-      })
+    // Prepare request
+    const cookie = await getCookieString(url)
+    const headers = Object.assign(opts.headers || {}, (cookie ? { cookie: cookie } : {}))
+    opts = Object.assign(opts, {
+      headers: headers
+    })
+
+    // Actual request
+    const res = await fetch(url, opts)
+
+    // Get cookie header
+    var cookies = []
+    if (res.headers.getAll) {
+      // node-fetch v1
+      cookies = res.headers.getAll('set-cookie')
+      // console.warn("You are using a fetch version that supports 'Headers.getAll' which is deprecated!")
```

Comment:
I added this warning but commented it out since it would be very annoying to have this in the console output every time.

I'd like to have this warning be displayed on package installment **if** the project has a certain node-fetch version that is deprecated... Currently too complicated and time consuming for me to implement.